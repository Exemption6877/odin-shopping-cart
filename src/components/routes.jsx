import App from "./App";
import Shop from "./Shop/Shop";
import Home from "./Home/Home";

const routes = [
  {
    path: "/",
    element: <App />,
    children: [
      { index: true, element: <Home /> },
      { path: "home", element: <Home /> },
      { path: "shop", element: <Shop /> },
    ],
  },
];

export default routes;
