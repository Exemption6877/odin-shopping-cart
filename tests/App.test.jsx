import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import routes from "../src/components/routes";

describe("App routing testing", () => {
  it("Renders default outlet content", () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ["/"],
    });

    const { asFragment } = render(<RouterProvider router={router} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders /shop outlet content", () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ["/shop"],
    });

    const { asFragment } = render(<RouterProvider router={router} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
