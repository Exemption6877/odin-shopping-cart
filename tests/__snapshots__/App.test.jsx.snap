// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component testing > Renders /shop outlet content 1`] = `
<DocumentFragment>
  <nav>
    Nav here
  </nav>
  <h1>
    Shop goes here
  </h1>
  <footer>
    Footer here
  </footer>
</DocumentFragment>
`;

exports[`App component testing > Renders default outlet content 1`] = `
<DocumentFragment>
  <nav>
    Nav here
  </nav>
  <h1>
    Home goes here
  </h1>
  <footer>
    Footer here
  </footer>
</DocumentFragment>
`;

exports[`App routing testing > Renders /shop outlet content 1`] = `
<DocumentFragment>
  <div
    class="_app_2e2173"
  >
    <nav
      class="_nav_2e2173"
    >
      Nav here
    </nav>
    <main
      class="_main_2e2173"
    >
      <h1>
        Shop goes here
      </h1>
    </main>
    <footer
      class="_footer_2e2173"
    >
      Footer here
    </footer>
  </div>
</DocumentFragment>
`;

exports[`App routing testing > Renders default outlet content 1`] = `
<DocumentFragment>
  <div
    class="_app_2e2173"
  >
    <nav
      class="_nav_2e2173"
    >
      Nav here
    </nav>
    <main
      class="_main_2e2173"
    >
      <div
        class="_imgBg_345077"
      >
        <h1>
          Home goes here
        </h1>
      </div>
    </main>
    <footer
      class="_footer_2e2173"
    >
      Footer here
    </footer>
  </div>
</DocumentFragment>
`;
